integration/tests/git_source/git_source_repeated_remote.rb

```diff
@@ -0,0 +1,68 @@
+require 'git_utils'
+require 'r10k_utils'
+require 'master_manipulator'
+test_name 'Verify the same remote can be used in more than one object'
+
+env_path = on(master, puppet('config print environmentpath')).stdout.rstrip
+r10k_fqp = get_r10k_fqp(master)
+git_environments_path = '/root/environments'
+git_repo_path = '/git_repos'
+git_repo_name = 'environments'
+git_control_remote = File.join(git_repo_path, "#{git_repo_name}.git")
+code_dir = '/etc/puppetlabs/code/environments/production'
+
+last_commit = git_last_commit(master, git_environments_path)
+git_provider = ENV['GIT_PROVIDER']
+r10k_config_path = get_r10k_config_file_path(master)
+r10k_config_bak_path = "#{r10k_config_path}.bak"
+#In-line files
+r10k_conf = <<-CONF
+cachedir: '/var/cache/r10k'
+git:
+  provider: '#{git_provider}'
+sources:
+  control:
+    basedir: "#{env_path}"
+    remote: "#{git_control_remote}"
+CONF
+
+# Install the same module in two different places
+puppetfile = <<-EOS
+mod 'prod_apache',
+  :git => 'git://github.com/puppetlabs/puppetlabs-apache.git',
+  :branch => 'master'
+
+mod 'test_apache',
+  :git => 'git://github.com/puppetlabs/puppetlabs-apache.git',
+  :branch => 'master'
```

Comment:
Would it be more realistic for this to be the 'test' branch?

Is it worth testing both ^the more realistic case and a case where the sources are literally identical? Or is that kind of redundant/unnecessary?

Reply:
That module only has a `master` and a `release` branch in its repo, and I'm not sure if release is permanent. I need to make sure it's something that this test is unlikely to fail on.

Reply:
Users in the r10k community were unable to come up with a _real_ use case that would trigger this bug. So no matter what we do here, it's not likely to match up to any known real workflow.

Reply:
gotcha, okay 👍

Reply:
I could find a different module that has more branches, or create a fixture, but we use `apache` elsewhere in these tests so I went with it.

Reply:
if we don't think the two cases are different, I'm fine leaving it